core/object/object.h

```diff
@@ -238,6 +238,46 @@ struct MethodInfo {
 ////else
 //return nullptr;
 
+// API used to extend in GDNative and other C compatible compiled languages
+class MethodBind;
+
```

Comment:
For those reviewing, most function pointers here are C compatible (they pass pointers to engine types though, but this is supported in GDNative), so the idea is to call passing this directly to GDNative.